README.md

```diff
@@ -1,14 +1,76 @@
-> **GETTING STARTED:** You should likely start with the `/mock` folder from your solution code for the mock gearup.
-
 # Project Details
 
+Project 3: Mock
+Team members: ejones38 jwu191
+Time Estimate: 15 hours
+Github link: https://github.com/cs0320-s24/mock-jwu191-ejones38.git
+
 # Design Choices
 
+## REPLFunction
+
+We chose to modify our repl function interface by adding additional parameters. We require any function implementing the interface to take in a boolean for the type of response, an array of string corresponding to its arguments, a string corresponding to the name of the file, and a boolean representing if the file has headers along with the setters for all of the above arguments. We chose to do so in order to garuantee that all of our functions have the required information to return the proper response. Additionally our repl function returns JSX.Elements instead of 2d array of strings. By returning JSX.ELements we our able to format our responses all in one step which making response generation much simpler. Additionally by returning JSX.Elements we our able to change our output between brief and verbose all in one step saving considerable effort.
+
+## History Type
+
+In this project we chose to have represent our history as a array of JSX.Elements. Since our functions all return JSX.Elements it made sense to just directly add those elements to our history instead of interconverting our outputs between a 2d array of strings and a JSX.Elemen
+
+## Handlers class
+
+All of our command functions are stored in the Handlers class. Each handler implements the REPLFunction interface taking in series of arguments and returning a JSX.Element By having all handlers contained in the same class it simplifies the addition of new commands
+
+## Command Selection
+
+Commands are input through a command line. We split this input into an array of arguments with the first argument being the command name itself. All commands alos exist in a map of key (command name) to value (commmand function). This allows us to take commandline input, separate it into its arguemtns, and pass the command name to our map returning the correct command function. This enables us to add and remove commands easily from our front-end program.
```

Comment:
How should a developer add their own function? What files/consts should they modify to implement their own action? How might they remove functions (i.e. load/search/view) if they don't want to use them? Also, this note makes it sound like you can create a new command or remove the functionality of an existing command from the front end REPL, which I don't think is accurate